### PR TITLE
fix: plugin API update event gets called repeatedly, missing hook deps

### PR DESCRIPTION
**Pull request by rot1024**
# Overview

## What I've done

- Fix plugin API update event gets called repeatedly
- Fix missing hook deps
   - Update eslint-config-reearth since the essential ESLint rules (`react-hooks/exhaustive-deps`) were missing for `.ts` files.
- The width of the right panel now changes when a layer is selected, and CSS has been added to prevent this.

## What I haven't done

- I found a bug in the camera limiter, but it has not been fixed.

## How I tested

- Manually locally. I confrimed plugins work

### CODE REVIEW

File: src/components/atoms/Resizable/index.tsx [hunk 1]
```diff
@@ -50,6 +50,7 @@ const StyledResizable = styled.div<Pick<Props, "direction" | "size">>`
   flex-direction: ${({ direction }) => (direction === "vertical" ? "row" : "column")};
   width: ${({ direction, size }) => (direction === "horizontal" ? null : `${size}px`)};
   height: ${({ direction, size }) => (direction === "vertical" ? null : `${size}px`)};
+  flex-shrink: 0;
```
Comment (rot1024) [inline src/components/atoms/Resizable/index.tsx:53]: Prevent a slight change in the width of the right panel when selecting layers

File: src/components/molecules/Visualizer/utils.ts [hunk 2]
```diff
@@ -11,12 +11,16 @@ export function mergeProperty(a: any, b: any) {
...
+  property: T | undefined,
 ): [T, (pluginId: string, property: T) => void] {
-  const [{ "": overriddenPropertyCommon, ...overriddenProperty }, setOverrideProperty] = useState<{
+  const [overriddenProperties, setOverrideProperty] = useState<{
     [pluginId: string]: any;
   }>({});
+  const { overriddenPropertyCommon, overriddenProperty } = useMemo(() => {
+    const { "": overriddenPropertyCommon, ...overriddenProperty } = overriddenProperties;
+    return { overriddenPropertyCommon, overriddenProperty };
+  }, [overriddenProperties]);
```
Comment (rot1024) [inline src/components/molecules/Visualizer/utils.ts:23]: This is where it happened: the rest spread syntax always returns a new object. This was causing endless rendering.

File: .eslintrc.yml [hunk 3]
```diff
@@ -2,6 +2,7 @@ extends:
   - reearth
 root: true
 rules:
+  # react-hooks/exhaustive-deps: warn
```
Comment (KaWaite) [inline .eslintrc.yml:5]: Is this suppose to be commented out?

**Review by KaWaite — APPROVED** [review]
Approved but please see one comment

**Pull request merged**